Docs/types.md

```diff
@@ -0,0 +1,645 @@
+	GroupMembershipAttributes []string `json:"groupMembershipAttributes"`
+
+	// AllUsersQuery holds the template for an LDAP query that returns user entries.
+	AllUsersQuery LDAPQuery `json:"usersQuery"`
+
+	// UserUIDAttribute defines which attribute on an LDAP user entry will be interpreted as its unique identifier.
+	// It must correspond to values that will be found from the GroupMembershipAttributes
+	UserUIDAttribute string `json:"userUIDAttribute"`
+
+	// UserNameAttributes defines which attributes on an LDAP user entry will be used, in order, as its OpenShift user name.
+	// The first attribute with a non-empty value is used. This should match your PreferredUsername setting for your LDAPPasswordIdentityProvider
+	UserNameAttributes []string `json:"userNameAttributes"`
+
+	// TolerateMemberNotFoundErrors determines the behavior of the LDAP sync job when missing user entries are
+	// encountered. If 'true', an LDAP query for users that doesn't find any will be tolerated and an only
+	// and error will be logged. If 'false', the LDAP sync job will fail if a query for users doesn't find
+	// any. The default value is 'false'. Misconfigured LDAP sync jobs with this flag set to 'true' can cause
+	// group membership to be removed, so it is recommended to use this flag with caution.
+	TolerateMemberNotFoundErrors bool `json:"tolerateMemberNotFoundErrors"`
+
+	// TolerateMemberOutOfScopeErrors determines the behavior of the LDAP sync job when out-of-scope user entries
+	// are encountered. If 'true', an LDAP query for a user that falls outside of the base DN given for the all
+	// user query will be tolerated and only an error will be logged. If 'false', the LDAP sync job will fail
+	// if a user query would search outside of the base DN specified by the all user query. Misconfigured LDAP
+	// sync jobs with this flag set to 'true' can result in groups missing users, so it is recommended to use
+	// this flag with caution.
+	TolerateMemberOutOfScopeErrors bool `json:"tolerateMemberOutOfScopeErrors"`
+}
+
+// ActiveDirectoryConfig holds the necessary configuration options to define how an LDAP group sync interacts with an LDAP
+// server using the Active Directory schema
+type ActiveDirectoryConfig struct {
+	// AllUsersQuery holds the template for an LDAP query that returns user entries.
+	AllUsersQuery LDAPQuery `json:"usersQuery"`
+
+	// UserNameAttributes defines which attributes on an LDAP user entry will be interpreted as its OpenShift user name.
+	UserNameAttributes []string `json:"userNameAttributes"`
+
+	// GroupMembershipAttributes defines which attributes on an LDAP user entry will be interpreted
+	// as the groups it is a member of
+	GroupMembershipAttributes []string `json:"groupMembershipAttributes"`
+}
+
+// AugmentedActiveDirectoryConfig holds the necessary configuration options to define how an LDAP group sync interacts with an LDAP
+// server using the augmented Active Directory schema
+type AugmentedActiveDirectoryConfig struct {
```

Comment:
Double check that LDAP identity provider doesn't use this somewhere. If not, remove along with other ldap group sync stuff